app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,355 @@
+
+    protected Group mGroup = null;
+    private RecyclerView mCardList;
+    private View mHelpText;
+    private View mNoMatchingCardsText;
+    private View mNoGroupCardsText;
+    private EditText mGroupNameText;
+    private TextView mGroupNameLabel;
+    private ActionBar mActionBar;
+
+    private HashMap<Integer, Boolean> mAdapterState;
+    private String mCurrentGroupName;
+
+    private boolean mGroupNameNotInUse;
+
+    private boolean mDarkMode;
+
+    @Override
+    protected void onCreate(Bundle inputSavedInstanceState)
+    {
+        super.onCreate(inputSavedInstanceState);
+        setContentView(R.layout.activity_manage_group);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        mHelpText = findViewById(R.id.helpText);
+        mNoMatchingCardsText = findViewById(R.id.noMatchingCardsText);
+        mNoGroupCardsText = findViewById(R.id.noGroupCardsText);
+        mCardList = findViewById(R.id.list);
+
+        mGroupNameText = findViewById(R.id.editTextGroupName);
+        mGroupNameLabel = findViewById(R.id.textViewEditGroupName);
+
+        mAdapter = new ManageGroupCursorAdapter(this, null, this);
+        mCardList.setAdapter(mAdapter);
+        registerForContextMenu(mCardList);
+
+        mGroup = null;
+
+        mDarkMode = Utils.isDarkModeEnabled(getApplicationContext());
+
+        if (inputSavedInstanceState != null) {
+            ManageGroupActivityInGroupState adapterState = inputSavedInstanceState.getParcelable("mAdapterState");
+            if (adapterState != null) {
+                mAdapterState = adapterState.getMap();
+            }
+            mCurrentGroupName = inputSavedInstanceState.getString("mCurrentGroupName");
+        }
+
+        mActionBar = getSupportActionBar();
+        if (mActionBar == null){
+            throw(new RuntimeException("mActionBar is not expected to be null here"));
+        }
+        mActionBar.setDisplayHomeAsUpEnabled(true);
+        mActionBar.setDisplayShowHomeEnabled(true);
+    }
+
+    private void resetGroupNameTextColor() {
+        if (mDarkMode) {
+            mGroupNameText.setTextColor(Color.WHITE);
+        } else{
+            mGroupNameText.setTextColor(Color.BLACK);
+        }
+    }
+
+    private void checkIfGroupNameIsInUse(){
+        mGroupNameNotInUse = false;
+        String currentGroupName = mGroupNameText.getText().toString();
+        if (!mGroup._id.equals(currentGroupName.trim())) {
+            Group group = mDB.getGroup(currentGroupName.trim());
+            if (group != null) {
+                mGroupNameNotInUse = false;
+                mGroupNameText.setTextColor(Color.RED);
+            } else {
+                mGroupNameNotInUse = true;
+            }
+        }
+    }
+
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+
+        Intent intent = getIntent();
+        mGroup = intent.getParcelableExtra("group");
```

Comment:
I'd much rather this activity receives a group ID in its intent, like LoyaltyCardViewActivity and LoyaltyCardEditActivity do.